src/rgw/CMakeLists.txt

```diff
@@ -57,6 +84,7 @@ set(librgw_common_srcs
   services/svc_user_rados.cc
   services/svc_zone.cc
   services/svc_zone_utils.cc
+  ${blake3_srcs}
```

Comment:
it doesn't seem right to add the blake3 sources directly to the rgw_common target. you've exposed the Blake3 digest in common/ceph_crypto.h but it wouldn't be useable outside of rgw

we should either link it into ceph-common, or split that out of common/ceph_crypto.h into a separate library with its own header, and have rgw depend on that

p.s. blake3 comes with its own [CMakeLists.txt](https://github.com/BLAKE3-team/BLAKE3/blob/master/c/CMakeLists.txt) that i'd hope we could use directly by putting `add_subdirectory(BLAKE3/c)` in our `src/CMakeLists.txt` and linking to its `BLAKE3::blake3` target

that should also set up the correct include directory so that we can `#include <blake3.h>` instead of `"BLAKE3/c/blake3.h"` which won't work if we ever switch from submodule to packages. on that note, i found blake3-devel packages in fedora/epel, but ubuntu/debian only package the rust library librust-blake3-dev

Reply:
I don't mind using their submodule, if it works out of the box.  I honestly think that these changes, could be in future commits, e.g., when another component wants to use blake3 or someone decides that platform integration with this blake3 implementation is now possible--it definitely would be a waste of effort now.

Reply:
(i.e., to worry about platform packages)